datauri/datauri.py

```diff
@@ -33,6 +48,21 @@ def __init__(self, media_type, data, uri):
         self.data = data
         self.uri = uri
 
+    @cached_property
+    def charset(self):
+        prefix = 'charset='
+        chunks = self.media_type.split(';')
+        for chunk in chunks:
+            if chunk.startswith(prefix):
+                return chunk[len(prefix):]
+        return None
+
+    @cached_property
+    def text(self):
+        if not self.media_type.startswith('text/'):
+            return None
+        return self.data.decode(self.charset or 'ascii')
```

Comment:
this will raise `LookupError` in case the encoding is a weird name like `foo`. should we error out or return `None` in that case?

Reply:
I think, something should be raised to allow lib users to make the decision about such cases themself.